File: spinnerCode/ProxH_lasso.m
% Proximity operator for function H, where
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%                                                                     %%%
%%%                H(D) := lambdaL*|| vec(W o D) ||_1                   %%%
%%%                                                                     %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%-------------------------------------------
%         Author:    Damian Brzyski
%         Date:      April 24, 2018
%-------------------------------------------
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%% 
function Dnew = ProxH_lasso(B, delta, W, lambdaL, WGTs)
    Bdelta     = B - W/delta;
    Dnew       = sign(Bdelta).*max(abs(Bdelta) - WGTs*lambdaL/delta, 0);
end

File: proxH_lasso.py
# Proximity operator for function H, where
###########################################################################
###                                                                   ###
###              H(D) := lambdaL*|| vec(W o D) ||_1                  ###
###                                                                    ###
###########################################################################

import numpy as np


def calc_proxH_lasso(B, delta, W, lambdaL, WGTs):
    Bdela =  B - W/delta
    change_matrix = abs(Bdela) - WGTs*lambdaL/delta
    change_matrix[change_matrix<0] = 0 # to mi zastępuje funcke max(x,0) w matlabie
    Dnew = np.sign(Bdela) * change_matrix
    return Dnew

#calc_proxH_lasso(Bnew, delta, Wk, lambdaL, W)

